### PR TITLE
Defines TypeScript types for rippled account method requests & responses

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR creates new TypeScript types for all rippled account methods (https://xrpl.org/account-methods.html), both the request shapes and response shapes.

### Context of Change

xrpl.js v2.0 models

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan

This PR doesn't change any existing code, so tests are fine.

### CODE REVIEW

File: src/models/common/index.ts [hunk 1]
```diff
@@ -11,10 +11,15 @@ export interface IssuedCurrency {
   issuer: string
 }
 
+export interface Balance {
+  currency: string
+  value: string
+}
+
```
Comment (ledhed2222) [inline src/models/common/index.ts:18]: Whta's a balance?

Comment (mvadari) [reply src/models/common/index.ts:18]: I made up a name for the object, happy to change it/move it. This type is used by `gateway_balances` and probably nowhere else.

Comment (ledhed2222) [reply src/models/common/index.ts:18]: if it's only used by `GatewayBalances` then shouldn't it be defined in some `gatewayBalances.ts` file and _not_ exported?

File: src/models/methods/account_channels.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    result: {
+        account: string
+        channels: Channel[]
+        ledger_hash: string
+        ledger_index: number
+        validated?: boolean
+        limit?: number
+        marker?: any
+    }
+}
```
Comment (ledhed2222) [inline src/models/methods/account_channels.ts:39]: we should be using camel case for file names in JS/TS

File: src/models/methods/account_currencies.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export interface AccountCurrenciesResponse extends BaseResponse {
+    result: {
+        ledger_hash?: string
+        ledger_index: number
+        receive_currencies: string[]
+        send_currencies: string[]
+        validated: boolean
+    }
+}
```
Comment (ledhed2222) [inline src/models/methods/account_currencies.ts:21]: camelcase file names

File: src/models/methods/account_info.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+export interface AccountCurrenciesResponse extends BaseResponse {
+  result: {
+    account_data: AccountRoot
+    signer_lists?: SignerList[]
+    ledger_current_index?: number
+    ledger_index?: number
+    queue_data?: QueueData
+    validated?: boolean
+  }
+}
```
Comment (ledhed2222) [inline src/models/methods/account_info.ts:41]: camelcase

File: src/models/methods/account_offers.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export interface AccountOffer {
+  flags: number
+  seq: number
+  taker_gets: Amount
+  taker_pays: Amount
+  quality: string
+  expiration?: number
+}
```
Comment (ledhed2222) [inline src/models/methods/account_offers.ts:21]: why is this one exported?

Comment (mvadari) [reply src/models/methods/account_offers.ts:21]: Unintentional, it's fixed now

File: src/models/common/transaction.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        LedgerIndex: string
+        NewFields: {[field: string]: any}
+    }
+}
+
+interface ModifiedNode {
+    ModifiedNode: {
+        LedgerEntryType: string
+        LedgerIndex: string
+        FinalFields: {[field: string]: any}
```
Comment (natenichols) [inline src/models/common/transaction.ts:15]: Is there any way we can narrow this from `{[field: string]: any`? Does this come in the form of a `LedgerEntry` that we can use to replace `any` here?

File: src/models/methods/baseMethod.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+interface Warning {
+    id: number
+    message: string
+    details?: {[key: string]: string}
+}
+
+export interface BaseResponse {
+    id: number | string
+    status: string
```
Comment (natenichols) [inline src/models/methods/baseMethod.ts:15]: Can status be anything other than `success` or `error`?

File: src/models/methods/baseMethod.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    type: string
+    result: any
+    warning?: "load"
+    warnings?: Warning[]
+    forwarded?: boolean
+    error?: string
+    request?: any
```
Comment (natenichols) [inline src/models/methods/baseMethod.ts:22]: Could this be type `BaseResponse`? IIRC, request just echos back the request sent to the server...

File: src/models/methods/baseMethod.ts (hunk 7 above)
Comment (mvadari) [reply src/models/methods/baseMethod.ts:15]: I'm not entirely sure, which is why I left it as `string` - maybe @intelliot knows

File: src/models/methods/baseMethod.ts (hunk 8 above)
Comment (mvadari) [reply src/models/methods/baseMethod.ts:22]: Good point

Comment (mvadari) [reply src/models/methods/baseMethod.ts:22]: No, it would actually have to be `Response`

File: src/models/common/transaction.ts (hunk 6 above)
Comment (mvadari) [reply src/models/common/transaction.ts:15]: You could in theory narrow it, but as far as I understand, it's a subset of `LedgerEntry` fields, and isn't necessarily a full `LedgerEntry` object. I suppose this is another question for @intelliot

**Review by natenichols — APPROVED** [review]
LGTM 👍 , given the two open questions for @intelliot are answered before merging.

File: src/models/common/transaction.ts (hunk 6 above)
Comment (intelliot) [reply src/models/common/transaction.ts:15]: This is probably any (modifiable) fields in any ledger object, so I believe there are a lot of possibilities, and it'll expand in the future as more kinds of ledger objects are added in the future. I would fine with leaving this as-is, but of course, if we want to make this more specific in the future, that would be fine as well.

File: src/models/methods/baseMethod.ts (hunk 7 above)
Comment (intelliot) [reply src/models/methods/baseMethod.ts:17]: I don't know. I think this is fine as-is, or `"success" | string` if you want to document the fact that 'success' is the happy path value

**Pull request merged**